src/app/routes/connector-ui/asset-page/asset-edit-dialog/asset-edit-dialog.component.html

```diff
@@ -180,6 +180,181 @@ <h1 mat-dialog-title>
+            </mat-form-field>
+
+            <button
+              mat-button
+              color="warn"
+              style="height: 54px; margin-top: 4px; margin-left: 8px"
+              (click)="form.onNutsLocationsRemoveClick(i)">
+              Remove
+            </button>
+          </div>
+
+          <div class="flex flex-row mb-[10px]">
+            <button
+              mat-button
+              color="accent"
+              (click)="form.onNutsLocationsAddClick()">
+              Add location
+            </button>
+          </div>
+
+          <!-- Data sample URLs -->
+          <div class="form-section-title">Data samples</div>
+          <div
+            *ngFor="
+              let sample of form.advanced.controls.dataSampleUrls.controls;
+              let i = index
+            "
+            class="flex flex-row space-x-[10px]">
+            <mat-form-field class="grow">
+              <mat-label>URL</mat-label>
+              <input
+                matInput
+                placeholder="https://example.com"
+                [formControl]="sample" />
+            </mat-form-field>
+
+            <button
+              mat-button
+              color="warn"
+              style="height: 54px; margin-top: 4px; margin-left: 8px"
+              (click)="form.onDataSampleUrlsRemoveClick(i)">
+              Remove
+            </button>
+          </div>
+
+          <div class="flex flex-row mb-[10px]">
+            <button
+              mat-button
+              color="accent"
+              (click)="form.onDataSampleUrlsAddClick()">
+              Add data sample
+            </button>
+          </div>
+
+          <!-- Reference file URLs -->
+          <div class="form-section-title">Reference files</div>
+          <div
+            *ngFor="
+              let file of form.advanced.controls.referenceFileUrls.controls;
+              let i = index
+            "
+            class="flex flex-row space-x-[10px]">
+            <mat-form-field class="grow">
+              <mat-label>URL</mat-label>
+              <input
+                matInput
+                placeholder="https://example.com"
+                [formControl]="file" />
+            </mat-form-field>
+
+            <button
+              mat-button
+              color="warn"
+              style="height: 54px; margin-top: 4px; margin-left: 8px"
+              (click)="form.onReferenceFileUrlsRemoveClick(i)">
+              Remove
+            </button>
+          </div>
+
+          <div class="flex flex-row mb-[20px]">
+            <button
+              mat-button
+              color="accent"
+              (click)="form.onReferenceFileUrlsAddClick()">
+              Add reference file
+            </button>
+          </div>
+
+          <!-- Reference files description -->
+          <mat-form-field
+            *ngIf="this.form.advanced.controls.referenceFileUrls.length > 0">
+            <mat-label>Reference files description</mat-label>
```

Comment:
could use the placeholder "..." and maybe a description text "Additional information regarding the reference files"